src/trie.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding utf-8 -*-
+
+END_OF_WORD = '$'
+
+
+class Node(object):
+    """Node object.  Has a value and a list of nodes that could come next."""
+
+    def __init__(self, value=None):
+        """
+        Each node will contain the letter at that node and a list of
+        node that come next.
+        """
+        self.value = value
+        self.next = []
```

Comment:
A list is not a good data structure to use for this purpose. The Trie data structure relies on quick and efficient lookup of stored values. To find something in a list, you have to iterate over the whole thing, which is a costly O(n) operation. There are other data structures available in Python which can look up an item in O(1) time -- use one of those instead!